corehq/apps/ota/models.py

```diff
@@ -89,3 +90,40 @@ def _delete_restore_blob(self):
         self.restore_blob_id = None
 
         return deleted
+
+
+class SerialIdBucket(models.Model):
+    """
+    Model used to keep track of an incrementing, unique integer
+    to be used in serial ID generation
+    """
+    domain = models.CharField(max_length=255)
+    bucket_id = models.CharField(max_length=255)
+    current_value = models.IntegerField(default=-1)
+
+    class Meta:
+        index_together = ('domain', 'bucket_id',)
+        unique_together = ('domain', 'bucket_id',)
+
+    @classmethod
+    def get_next(cls, domain, bucket_id, session_id=None):
+        if session_id is not None:
```

Comment:
Whats the reason for using this instead of just `if session_id:`?

Reply:
I was thinking of it as being more explicitly - that is, if no argument was passed in for `session_id`, skip the cache.  However, I suppose providing an empty string would lead to more confusing behavior, so I'll switch it.